lib/src/subtree.c

```diff
@@ -940,27 +946,28 @@ static size_t ts_subtree__write_to_string(
     }
   }
 
-  if (is_visible) cursor += snprintf(*writer, limit, ")");
+  if (is_visible || (is_root && ts_subtree_child_count(self) > 0)) cursor += snprintf(*writer, limit, ")");
 
   return cursor - string;
 }
 
 char *ts_subtree_string(
   Subtree self,
+  TSSymbol alias_symbol,
   const TSLanguage *language,
   bool include_all
 ) {
   char scratch_string[1];
   size_t size = ts_subtree__write_to_string(
     self, scratch_string, 1,
     language, include_all,
-    0, false, ROOT_FIELD
+    alias_symbol, false, ROOT_FIELD
```

Comment:
It looks like, in addition to passing in the `alias_symbol` here, we need to look up whether the alias symbol represents a "named symbol", similarly to what we do in the recursive calls to this function

```c
        bool subtree_alias_is_named = subtree_alias_symbol
          ? ts_language_symbol_metadata(language, subtree_alias_symbol).named
          : false;
```

We need to pass in that boolean, instead of always passing in `false`. I think that if you did that, then the `is_visible` flag would always be correct, and you wouldn't need a lot of the other logic that you added to the `ts_subtree__write_to_string` function, in order to work around `is_visible` being wrong.